src/Avalonia.Animation/Animator`1.cs

```diff
@@ -28,21 +28,25 @@ public abstract class Animator<T> : AvaloniaList<AnimatorKeyFrame>, IAnimator
         public Animator()
         {
             // Invalidate keyframes when changed.
-             this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
+            this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
         }
 
         /// <inheritdoc/>
         public virtual IDisposable Apply(Animation animation, Animatable control, IObservable<bool> match, Action onComplete)
         {
-             if (!_isVerifiedAndConverted)
+            if (!_isVerifiedAndConverted)
                 VerifyConvertKeyFrames();
 
             return match
-                .Where(p => p)
-                .Subscribe(_ =>
-                {
-                    var timerObs = RunKeyFrames(animation, control, onComplete);
-                });
+                     .DistinctUntilChanged()
+                     .Select(x => x ? RunKeyFrames(animation, control, onComplete) : null)
+                     .Buffer(2, 1)
+                     .Where(x => x.Count > 1)
+                     .Subscribe(x =>
+                        {
+                            if (animation.RepeatCount.IsLoop)
```

Comment:
Don't we want to cancel non-looping animations when the selector no longer matches?

Reply:
Right, i've double-checked with CSS and it does cancel animations when the selector turns false